mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -121,8 +131,9 @@ class LinearRegression(override val uid: String)
       })
 
     val numFeatures = summarizer.mean.size
-    val yMean = statCounter.mean
+    val yMean = if ($(fitIntercept)) statCounter.mean else 0.0
```

Comment:
I think passing $(fitIntercept) into `LeastSquaresAggregator` to set `offset` as zero will be cleaner. I will send out another PR proving that `offset` zero is the same optimizing without intercept.